tests/rptest/tests/datalake/datalake_e2e_test.py

```diff
@@ -44,3 +77,49 @@ def test_e2e_basic(self, query_engine, filesystem_catalog_mode):
             dl.create_iceberg_enabled_topic(self.topic_name)
             dl.produce_to_topic(self.topic_name, 1024, count)
             dl.wait_for_translation(self.topic_name, msg_count=count)
+
+    @cluster(num_nodes=4)
+    @matrix(query_engine=[QueryEngineType.SPARK, QueryEngineType.TRINO])
+    def test_avro_schema(self, query_engine):
+        count = 100
+        table_name = f"redpanda.{self.topic_name}"
+
+        with DatalakeServices(self.test_ctx,
+                              redpanda=self.redpanda,
+                              filesystem_catalog_mode=True,
+                              include_query_engines=[query_engine]) as dl:
+            dl.create_iceberg_enabled_topic(self.topic_name)
+            avro_serde_client = self._get_serde_client(SchemaType.AVRO,
+                                                       SerdeClientType.Golang,
+                                                       self.topic_name, count)
+            avro_serde_client.start()
+            avro_serde_client.wait()
+            dl.wait_for_translation(self.topic_name, msg_count=count)
+
+            if query_engine == QueryEngineType.TRINO:
+                trino = dl.trino()
+                trino_expected_out = [('redpanda_offset', 'bigint', '', ''),
+                                      ('redpanda_timestamp', 'timestamp(6)',
+                                       '', ''),
+                                      ('redpanda_key', 'varbinary', '', ''),
+                                      ('redpanda_value', 'varbinary', '', ''),
+                                      ('payload', 'row(val bigint)', '', '')]
+                trino_describe_out = trino.run_query_fetch_all(
```

Comment:
for future: this looks like a good candidate to push into the base with some sort of type mapping from iceberg types to engine native types.

Reply:
100% I mulled over this a bit and was thinking maybe it'd be good to expose the table info through some dataframe interface, but it adds another layer of translation.. So I've also landed on having something more explicit with regards to SQL types, though haven't come to conclusions about what it'll look like, especially for nested types